packages/editor/src/components/global-styles-provider/index.js

```diff
@@ -146,13 +146,10 @@ function useGlobalStylesUserConfig() {
 
 function useGlobalStylesBaseConfig() {
 	const baseConfig = useSelect( ( select ) => {
-		const { __experimentalGetCurrentThemeBaseGlobalStyles, canUser } =
+		const { __experimentalGetCurrentThemeBaseGlobalStyles } =
 			select( coreStore );
 
-		return (
-			canUser( 'read', { kind: 'root', name: 'theme' } ) &&
```

Comment:
Something I just noticed, this is only checking whether the user can read the `theme`. While the `__experimentalGetCurrentThemeBaseGlobalStyles` does get the current theme, there's nothing to check whether the user can read the `globalStyles` entity (the actual entity type being returned. 🤔

Reply:
How our permissions checks are supposed to work with entities is going over my head at the moment 🫠 

The REST API endpoint called when retrieving this base global styles data does check if the user has the `edit_posts` capability in this PR, or `edit_theme_options` on trunk. So is that not "something"?

Reply:
When debugging I find it helpful to remove the preloading so that you can see the requests in your dev tools. The permissions checks from `canUser` are generally `OPTIONS` requests, and the `Allow` header in the response provides a list of the HTTP verbs that the current user is allowed to perform.

The `canUser` call you deleted here doesn't seem to work from what I can tell. I think that's because it doesn't match the actual call that's made to get the current theme. My understanding is that `canUser( 'read', { kind: 'root', name: 'theme' } )` is checking "can the user `GET` a list of installed themes" (which I'd expect lower tiered users not to be able to do). The actual request to get the current theme is querying for `status: active`:
https://github.com/WordPress/gutenberg/blob/482c18803ed9becccc57d1b23632438e1afc1d67/packages/core-data/src/resolvers.js#L374-L378

It involves different permissions in the REST Controller:
https://github.com/WordPress/wordpress-develop/blob/f4761a3f8c5aa6920d4407e3af271ef70db5f305/src/wp-includes/rest-api/endpoints/class-wp-rest-themes-controller.php#L103-L106

The user only needs `edit_posts` caps:
https://github.com/WordPress/wordpress-develop/blob/f4761a3f8c5aa6920d4407e3af271ef70db5f305/src/wp-includes/rest-api/endpoints/class-wp-rest-themes-controller.php#L149-L152

I'll have a look into whether there's a way to replace that with a check that actually works.

I think there's a similar issue with trying to do `canUser( 'read', { kind: 'root', name: 'globalStyles' } )`. That would check whether the user can list all global styles, but we actually need to check a route that's more like `globalStyles/theme/<activeStylesheet>`.

Reply:
I've pushed a commit that adds working permissions checks.

Reply:
So if the approach in https://github.com/WordPress/gutenberg/pull/64797/commits/790febdf67899a8d659dd894b5cadeec982f4135 is going back towards checking capabilities directly as before https://github.com/WordPress/gutenberg/pull/63812. Does that mean we can also roll back the changes to the global styles entity definition?

Reply:
If pushing changes to this PR, it might also helps those pinged for input if the PR description is updated to match new changes.

Reply:
Actually, seems like I was logged in as admin accidentally when I thought it was working. It doesn't work with non-admin so I'll revert those commits.

The explanation is still correct, but unfortunately it doesn't look like there's a way to check permissions using `canUser` for those requests.